dependencies.yaml

```diff
@@ -744,6 +744,28 @@ dependencies:
           - pytest<8
           - pytest-cov
           - pytest-xdist
+    specific:
+      # Define additional constraints for testing with oldest dependencies.
+      - output_types: [conda, requirements]
+        matrices:
+          - matrix: {dependencies: "oldest"}
+            packages:
+              - numba==0.57.*
+              - numpy==1.23.*
+              - pandas==2.0.*
+              - pyarrow==16.1.0
```

Comment:
@mroeschke can we update this once #16681 merges?

Reply:
Updated to 14.0.0